myia/abstract/prim.py

```diff
@@ -0,0 +1,1022 @@
+            raise MyiaTypeError('Wrong class inst')
+
+    return AbstractClass(
+        cls.tag,
+        {
+            name: elem
+            for (name, _), elem in zip(expected, elems)
+        },
+        cls.methods
+    )
+
+
+@standard_prim(P.tuple_getitem)
+async def _inf_tuple_getitem(engine, arg: AbstractTuple, idx: dtype.Int[64]):
+    idx_v = idx.values[VALUE]
+    if idx_v is ANYTHING:
+        raise MyiaTypeError(
+            'Tuples must be indexed with a constant'
+        )
+    nelems = len(arg.elements)
+    if not -nelems <= idx_v < nelems:
+        raise MyiaTypeError(
+            'Tuple element out of range'
+        )
+    return arg.elements[idx_v]
+
+
+@standard_prim(P.list_getitem)
+async def _inf_list_getitem(engine, arg: AbstractList, idx: dtype.Int[64]):
+    return arg.element
+
+
+@standard_prim(P.tuple_setitem)
+async def _inf_tuple_setitem(engine,
+                             arg: AbstractTuple,
+                             idx: dtype.Int[64],
+                             value: AbstractBase):
+    idx_v = idx.values[VALUE]
+    if idx_v is ANYTHING:
+        raise MyiaTypeError(
+            'Tuples must be indexed with a constant'
+        )
+    nelems = len(arg.elements)
+    if not -nelems <= idx_v < nelems:
+        raise MyiaTypeError(
+            'Tuple element out of range'
+        )
+    elts = arg.elements
+    new_elts = tuple([*elts[:idx_v], value, *elts[idx_v + 1:]])
+    return AbstractTuple(new_elts)
+
+
+@standard_prim(P.list_setitem)
+async def _inf_list_setitem(engine,
+                            arg: AbstractList,
+                            idx: dtype.Int[64],
+                            value: AbstractBase):
+    engine.abstract_merge(arg.element, value)
+    return arg
+
+
+@standard_prim(P.list_append)
+async def _inf_list_append(engine,
+                           arg: AbstractList,
+                           value: AbstractBase):
+    engine.abstract_merge(arg.element, value)
+    return arg
+
+
+class _GetAttrInferrer(Inferrer):
+    async def run(self, engine, outref, argrefs):
+        check_nargs(P.getattr, 2, argrefs)
+        r_data, r_item = argrefs
+        data = await r_data.get()
+        item = await r_item.get()
+
+        def chk(data_v, item_v):
+            if not isinstance(item_v, str):  # pragma: no cover
+                raise MyiaTypeError(
+                    f'item argument to resolve must be a string, not {item_v}.'
+                )
+
+        async def on_dcattr(data, data_t, item_v):
+            return data.attributes[item_v]
+
+        rval = await static_getter(
+            engine, data, item,
+            fetch=getattr,
+            on_dcattr=on_dcattr,
+            chk=chk,
+            outref=outref,
+            dataref=r_data,
+        )
+        self.cache[(data, item)] = rval
+        return rval
+
+
+abstract_inferrer_constructors[P.getattr] = _GetAttrInferrer.partial()
+
+
+# setattr = Primitive('setattr')
```

Comment:
Is this a leftover?

Reply:
Yes. Good catch.

Reply:
Although, there's no inferrer for `setattr` yet, so I might leave a comment instead to mark the spot.

Reply:
I'm ok with that.